File: trees/binary_trees/search_bst.rb
def search_bst(root, val)
    if root.val == val
        root
    elsif root.val > val && root.left
        search_bst(root.left, val)
    elsif root.val < val && root.right
        search_bst(root.right, val)
    else
        []
    end
end

File: trees/binary_trees/search_bst.js
var searchBST = function(root, val) {
  if (root.val === val){
    return root
  } else if (root.left && root.val > val) {
      return searchBST(root.left, val)
  } else if (root.right && root.val < val) {
      return searchBST(root.right, val)
  } else {
    return []
  }
};
